lib/floe/workflow/runner/kubernetes.rb

```diff
@@ -4,42 +4,75 @@ module Floe
   class Workflow
     class Runner
       class Kubernetes < Floe::Workflow::Runner
-        attr_reader :namespace, :server, :token
+        attr_reader :namespace, :client
 
         def initialize(options = {})
           require "awesome_spawn"
           require "securerandom"
           require "base64"
+          require "kubeclient"
+          require "pathname"
           require "yaml"
 
           @namespace = options.fetch("namespace", "default")
-          @server    = options.fetch("server", nil)
-          @token     = options.fetch("token", nil)
-          @token   ||= File.read(options["token_file"]) if options.key?("token_file")
+
+          token     = options.fetch("token", nil)
+          token   ||= File.read(options["token_file"]) if options.key?("token_file")
+
+          kubeconfig_file = File.join(Dir.home, ".kube", "config")
+          kubeconfig      = Kubeclient::Config.read(kubeconfig_file) if File.exist?(kubeconfig_file)
+
+          api_endpoint = !kubeconfig.nil? ? kubeconfig.context.api_endpoint : options.fetch("server")
+          api_version  = options.fetch("api_version", "v1")
+          ssl_options  = !kubeconfig.nil? ? kubeconfig.context.ssl_options : {:ca_file => options.fetch("ca_file", nil), :verify_ssl => options.fetch("verify_ssl", OpenSSL::SSL::VERIFY_PEER)}
+          auth_options = !kubeconfig.nil? ? kubeconfig.context.auth_options : {:bearer_token => token}
```

Comment:
Minor, but I think there should be rearranged slightly to group together the kubeconfig based things:

```suggestion
          if kubeconfig
            api_endpoint = kubeconfig.context.api_endpoint
            ssl_options  = kubeconfig.context.ssl_options
            auth_options = kubeconfig.context.auth_options
          else
            api_endpoint = options.fetch("server")
            ssl_options = {:ca_file => options.fetch("ca_file", nil), :verify_ssl => options.fetch("verify_ssl", OpenSSL::SSL::VERIFY_PEER)}
            auth_options = {:bearer_token => token}
          end
          api_version  = options.fetch("api_version", "v1")
```

Reply:
I'm still playing around with this so expect this to change

Reply:
Okay updated, what I wanted to achieve is always allowing the user to override the environment via command-line options so we prioritize e.g. `options.fetch("server")` over `kubeconfig.context.api_endpoint`, otherwise with the above proposal if you had a `kubeconfig` file you would never be able to connect to anything else.

So current logic is as follows:
1. Command-line options
2. `/run/secrets/kubernetes.io/serviceaccount/token` if it exists
3. `~/.kube/config` if it exists
4. Raise exception due to missing info